processing/data_collection/gazette/spiders/rn_natal.py

```diff
@@ -0,0 +1,36 @@
+from datetime import datetime
+
+import dateparser
+from scrapy import Request, FormRequest
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class RnNatalSpider(BaseGazetteSpider):
+
+    name = "rn_natal"
+    allowed_domains = ["www.natal.rn.gov.br"]
+
+    def start_requests(self):
+        base_url = "http://www.natal.rn.gov.br/dom/"
+        for year in range(2003, datetime.now().year + 1):
+            for month in range(1, 13):
```

Comment:
If you use `dateutil.rrule` you can express your intent much more and have more power on the upper and lower date limits. [Here is the examples page](https://dateutil.readthedocs.io/en/stable/examples.html#rrule-examples), look for `MONTHLY`.

Reply:
I appreciate the link. This is a cool function, although I think `range` plays fine here. It's not something very complex.

Reply:
Let me know if you believe it should be changed anyway.

Reply:
I agree that it is not very complex, but `rrule` is a nicer interface regarding generating a date range.

If we could agree that following coding patterns in a project is good and we asked ourselves "what should be the coding pattern for date ranges in this project?", do you think that we should not use `rrule` and use `range`?

Reply:
Indeed, it's a nice interface.

But the question "should we make it a code pattern in this project?" should be addressed in another place as it's out of scope here. Perhaps we should raise an issue, tag it as a Question, and discuss the pros and cons there. For now, this project has 300+ spiders and it seems like only 3 of them are making use of `rrule` and only 2 of them are making use of the `AVAILABLE_FROM` constant. 

Questions that should be addressed before making it a pattern:

- how are we dealing with datetime imports?
  - `import datetime`
  - `import datetime as dt`
  - `from datetime import date, datetime`
- will `rrate` make it more difficult for first-time contributors or beginners?

Reply:
Those are good points. Would you open the issue regarding this questions?

Just a correction: the project has 300+ spiders but most of Santa Catarina are based on the `FecamGazetteSpider`. We are talking about a 30-40 spiders project.

I am pointing this out because I think now is a good time to be raising this type of questions and defining some kind of guideline/style before it gets too hard to refactor.

I will consider this comment out of scope and approve this PR.

Reply:
https://github.com/okfn-brasil/diario-oficial/issues/193